binary_search_tree/binary_search_tree.py

```diff
@@ -4,33 +4,51 @@ def __init__(self, value):
         self.left = None
         self.right = None
 
-    # Why can I not reference self in start?
-    def insert(self, value, start=self.value):
-        if start is not None:
-            # Moves it to the right if value is larger
-            if value > self.value:
-                if self.right is None:
-                    self.right = BinarySearchTree(value)
-                    return value
-                else:
-                    BinarySearchTree.insert(value, self.right)
-            # Moves to the left
-            else:
-                if self.left is None:
-                    self.left = BinarySearchTree(value)
-                    return value
-                else:
-                    BinarySearchTree.insert(value, self.left)
+    def insert(self, value):
+        direction = 'right' if value > self.value else 'left'
+        if getattr(self, direction) is not None:
```

Comment:
nice clever way to reference self directly. It techincally doesn't have an explicit direction value but it is either left or right. Good job Drew.